adapter/pkg/config/parser.go

```diff
@@ -66,25 +87,27 @@ func GetMgwHome() string {
 // from where the executable is called from.
 //
 // Returns the log configuration object mapped from the configuration file during the startup.
-func ReadLogConfigs() (*LogConfig, error) {
+func ReadLogConfigs() *LogConfig {
 	onceLogConfigRead.Do(func() {
-		adapterLogConfig = new(LogConfig)
-		_, err := os.Stat(GetMgwHome() + relativeLogConfigPath)
-		if err != nil {
-			logger.Fatal("Log configuration file not found.", err)
-			panic(err)
-		}
-		content, readErr := ioutil.ReadFile(mgwHome + relativeLogConfigPath)
-		if readErr != nil {
-			logger.Fatal("Error reading log configurations. ", readErr)
-			panic(err)
-		}
-		parseErr := toml.Unmarshal(content, adapterLogConfig)
-		if parseErr != nil {
-			logger.Fatal("Error parsing the log configuration ", parseErr)
-			panic(parseErr)
+		adapterLogConfig = getDefaultLogConfig()
+		filePath, err := GetLogConfigPath()
+		if err == nil {
+			content, readErr := ioutil.ReadFile(filePath)
+			if readErr != nil {
+				logger.Fatal("Proceeding with default log configuration as error occured while reading log configurations ", readErr)
+			} else {
+				parseErr := toml.Unmarshal(content, adapterLogConfig)
+				if parseErr != nil {
+					logger.Fatal("Proceeding with default log configuration as error occured while parsing the log configuration ", parseErr)
```

Comment:
Fatal will exit the process without continuing with the default configs?

Reply:
resolved via https://github.com/wso2/product-microgateway/pull/2100/commits/d40f354903cd9fae24039f4a8a123f162a5930a2